### PR TITLE
Add funding graph

### LINKED ISSUE #97: Have a graph showing funding over time
**Please describe the problem you're having.**
User's often ask about what the current funding rate is and how they can see past funding rates

**Screenshots**
Applications like dydx show graphs for historical funding 

https://user-images.githubusercontent.com/13340486/150662942-17e6d25b-3641-4a45-b179-4b9aaa13bea2.mov

**Additional context**
Could add this graph as a tab on the trade page next to the payoff, risks etc. sections

### PR DESCRIPTION
**Pull request by block902**
# Task: Add funding graph

## Description

Add historical funding graph to trade page

Fixes #97  (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Tested locally

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

File: packages/frontend/src/components/Charts/FundingChart.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  const classes = useStyles()
+
+  const normFactors: NormHistory[] = useNormHistory()
+  const graphData = normFactors.map((item) => {
+    const secondsElapsed = Number(item.timestamp) - Number(item.lastModificationTimestamp)
+    const deltaT = secondsElapsed / (420 * 60 * 60)
+    const markIndex = 1 / Math.exp(Math.log(Number(item.newNormFactor) / Number(item.oldNormFactor)) / deltaT)
+    const dayFunding = Math.log(markIndex) / 17.5
+    const monthFunding = dayFunding * 30
+    const yearFunding = dayFunding * 365.25
```
Comment (alexisgauba) [inline packages/frontend/src/components/Charts/FundingChart.tsx:53]: wait lol is it 365.25 to account for leap years?

**Review by alexisgauba — CHANGES_REQUESTED** [review]
Very cool! If we could add a hover that shows the val like we do for the backtest that would be great 

Screen Shot 2022-02-21 at 6 07 51 PM

In terms of ordering if funding could go after payoff and before risks that would be perfect. 

We should also add this to the short side, again after payoff and before risks 

Also would be good to show the values in percentages, looks like rn for example the graph is showing 0.001070 when current implied funding is .10%. 

Could we add a tooltip next to funding, maybe similar to how we have one next to historical days, that explains how funding is calculated? Same for vol 

Screen Shot 2022-02-21 at 6 12 08 PM

File: packages/frontend/src/components/Charts/FundingChart.tsx (hunk 1 above)
Comment (block902) [reply packages/frontend/src/components/Charts/FundingChart.tsx:53]: Yes, I think so. Joe calculated yearly funding in that way in the doc.

**Comment by block902** [comment]
> Very cool! If we could add a hover that shows the val like we do for the backtest that would be great
> 
> Screen Shot 2022-02-21 at 6 07 51 PM
> 
> In terms of ordering if funding could go after payoff and before risks that would be perfect.
> 
> We should also add this to the short side, again after payoff and before risks
> 
> Also would be good to show the values in percentages, looks like rn for example the graph is showing 0.001070 when current implied funding is .10%.
> 
> Could we add a tooltip next to funding, maybe similar to how we have one next to historical days, that explains how funding is calculated? Same for vol
> 
> Screen Shot 2022-02-21 at 6 12 08 PM

Hi Alexis. We need to put formula to calculate daily/monthly/annual funding and vol in the tooltip? It would be great if you can give me the text for the tooltips. And we need to show Vol in percent as well?

**Comment by block902** [comment]
> Very cool! If we could add a hover that shows the val like we do for the backtest that would be great
> 
> Screen Shot 2022-02-21 at 6 07 51 PM
> 
> In terms of ordering if funding could go after payoff and before risks that would be perfect.
> 
> We should also add this to the short side, again after payoff and before risks
> 
> Also would be good to show the values in percentages, looks like rn for example the graph is showing 0.001070 when current implied funding is .10%.
> 
> Could we add a tooltip next to funding, maybe similar to how we have one next to historical days, that explains how funding is calculated? Same for vol
> 
> Screen Shot 2022-02-21 at 6 12 08 PM

Made updates according to your comments and pushed commits. Please check again and let me know if it looks fine. Thanks.

File: packages/frontend/src/components/Charts/FundingChart.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    ...graphOptions,
+    localization: {
+      priceFormatter: (num: number) => `${num < 0 ? num.toFixed(3) : num.toFixed(4)}%`,
+    },
+  }
+  const startTimestamp = normFactors.length > 0 ? Number(normFactors[0].timestamp) : undefined
+  const endTimestamp = normFactors.length > 0 ? Number(normFactors[normFactors.length - 1].timestamp) : undefined
+  const legendText =
+    fundingType.id === 'vol'
+      ? 'Annual Vol (rhs)'
```
Comment (alexisgauba) [inline packages/frontend/src/components/Charts/FundingChart.tsx:122]:
```suggestion
      ? 'Annual Vol'
```

File: packages/frontend/src/constants/enums.ts [hunk 3]
```diff
@@ -71,6 +71,10 @@ export enum Tooltips {
   StrategyEarnFunding = 'You earn funding by depositing into the strategy',
   StrategyProfitThreshold = 'Based on current funding, crab strategy would be unprofitable if ETH moves more than approximately the profit threshold in either direction each day.',
   FullcloseInput = 'Select partial close to edit input',
+  FundingVol = 'VOL is calculated as "sqrt(Daily Funding * 365)"',
+  FundingDaily = 'Daily Funding is calculated as "secondsElapsed = current Norm History timestamp - last Norm History timestamp; deltaT = secondsElapsed / secondsElapsed / (420 * 60 * 60); markIndex = 1 / exp(ln(current Norm Factor value / old Norm Factor value) / deltaT); dailyFunding = ln(markIndex) / 17.5"',
```
Comment (alexisgauba) [inline packages/frontend/src/constants/enums.ts:75]:
```suggestion
  FundingDaily = 'Daily funding is calculated as ln(mark / index) * 17.5. Mark and Index spot values are taken at points in the past and scaled to one day. 
',
```

Comment (alexisgauba) [reply packages/frontend/src/constants/enums.ts:75]: thoughts on this @alpinechicken ?

Comment (alexisgauba) [reply packages/frontend/src/constants/enums.ts:75]: monthly, annual, and vol lgtm

**Review by alexisgauba — COMMENTED** [review]
Overall lgtm! Would def want to get @alpinechicken thoughts on the calcs

**Review by alpinechicken — COMMENTED** [review]
Calcs and charts lgtm

**Review by alpinechicken — APPROVED** [review]

**Review by alexisgauba — APPROVED** [review]

**Pull request merged**